pos_show_clock/static/src/js/ChromeWidgets/Clock.js

```diff
@@ -0,0 +1,42 @@
+odoo.define("pos_show_clock.Clock", function (require) {
+    "use strict";
+
+    const PosComponent = require("point_of_sale.PosComponent");
+    const Registries = require("point_of_sale.Registries");
+    const {useState} = owl;
+
+    class Clock extends PosComponent {
+        setup() {
+            super.setup();
+            this.state = useState({current_time: ""});
+            this.mounted();
+        }
+        mounted() {
+            this.updateCurrentTime();
+        }
+        updateCurrentTime() {
+            const now = Date.now();
+            const next_update = 60050 - parseInt(moment(now).format("ss")) * 1000;
+            // 1 min = 60.000ms
+            // 60000 + 50 (less then 100ms rule)
+            // it's the time we can update the minute without the user noticing
+            // the difference when comparing it to another watch.
+            this.state.current_time = moment(now).format("HH:mm");
+            this.invervalReference = setInterval(() => {
+                this.updateCurrentTime();
+            }, next_update);
+        }
+        unmount() {
+            clearInterval(this.intervalReference);
+        }
+        get time() {
+            return this.state.current_time;
+        }
```

Comment:
Maybe I wasn't clear about setInterval. It's not like setTimeout, you only have to set it up once, and it will automatically run the callback on each interval, forever.. until you clearit. That's why I suggest to set it up when the widget is mounted, and then clearit when it's unmounted.

Some other things:
- you can't call `mounted` on `setup`. `mounted` is an official lifecycle hook for components, you need to let the owl framework call it when it's actually being mounted.
- the unmount hook is `willUnmount` (I always forget that name)

I took the liberty of explaining with code, here below.
I also changed a few things, like I remove the need to store the formated time in a state. I haven't tested it, but see if it works and if it makes sense for you ^^

I do not understand the 60050 rule, so I made it update every second (`1000` ms, see `refreshInterval`). Anyway, feel free to take what you want or simply ignore.

```suggestion
        mounted() {
            this.intervalRef = setInterval(() => this.render(), this.refreshInterval);
        }
        willUnmount() {
            if (this.intervalRef) {
                clearInterval(this.intervalRef);
            }
        }
        get refreshInterval() {
            return 1000;
        }
        get timeFormat() {
            return "HH:mm";
        }
        get time() {
            return moment(now).format(this.timeFormat);
        }
```